File: QE-Selenium-Java/Activity12.java
package demos;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity12 {
    public static void main(String[] args) {

        FirefoxDriver driver = new FirefoxDriver();

        Actions actions = new Actions(driver);
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://training-support.net/webelements/dynamic-content");
        System.out.println(driver.getTitle());

        WebElement clickMeButton = driver.findElement(By.id("genButton"));
        actions.click(clickMeButton).build().perform();

        WebElement word_text = wait.until(
                ExpectedConditions.visibilityOfElementLocated(By.id("word"))
        );

        wait.until(
                ExpectedConditions.textToBePresentInElementLocated(By.id("word"), "release")
        );

        System.out.println(word_text.getText());

        driver.quit();
    }
}

File: QE-Selenium-Python/Activity12.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.common.action_chains import ActionChains

driver = webdriver.Firefox()

actions = webdriver.ActionChains(driver)
wait = WebDriverWait(driver, 10)
driver.get("https://training-support.net/webelements/dynamic-content") 
print(driver.title)
clickMeButton = driver.find_element(By.ID, "genButton")
actions.click(clickMeButton).perform()
word_text=wait.until(EC.visibility_of_element_located((By.ID, "word")))
wait.until(EC.text_to_be_present_in_element((By.ID, "word"), "release"))
print(word_text.text)
driver.quit()